R/learner_qgam_regr_qgam.R

```diff
@@ -0,0 +1,142 @@
+#' @title Regression Quantile Generalized Additive Model Learner
+#' @author lona-k
+#' @name mlr_learners_regr.qgam
+#'
+#' @description
+#' Quantile Regression with generalized additive models.
+#' Calls [qgam::qgam()] from package \CRANpkg{qgam}.
+#'
+#' @section Form:
+#' For the `form` parameter, a gam formula specific to the [Task][mlr3::Task] is required
+#' (see example and `?mgcv::formula.gam`).
+#' If no formula is provided, a fallback formula using all features in the task is used that will make the Learner
+#' behave like Linear Quantile Regression.
+#' Only features specified in the formula are used, superseding columns with col_roles "feature" in the task.
+#'
+#' @section Quantile:
+#' The quantile for the Learner, i.e. `qu` parameter from [qgam::qgam()],
+#' is set using the value specified in `learner$quantiles`.
+#'
+#' @templateVar id regr.qgam
+#' @template learner
+#'
+#' @references
+#' `r format_bib("faisolo2017qgam")`
+#'
+#' @template seealso_learner
+#' @examplesIf requireNamespace("qgam", quietly = TRUE)
+#' # simple example
+#' t = mlr3::tsk("mtcars")
+#' l = mlr3::lrn("regr.qgam")
+#' l$param_set$values$form = mpg ~ cyl + am + s(disp) + s(hp)
+#' l$quantiles = 0.25
+#' l$train(t)
+#' l$model
+#' l$predict(t)
+#' @export
+LearnerRegrQGam = R6Class("LearnerRegrQGam",
+  inherit = LearnerRegr,
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      param_set = ps(
```

Comment:
Format the parameter set with spaces like https://github.com/mlr-org/mlr3learners/blob/main/R/LearnerClassifXgboost.R